craft_providers/lxd/installer.py

```diff
@@ -96,6 +96,26 @@ def is_initialized(*, remote: str, lxc: LXC) -> bool:
     if not devices:
         return False
 
+    projects = lxc.project_list(remote=remote)
+    if "rockcraft" in projects:
+        devices = lxc.profile_show(
+            project="rockcraft", profile="default", remote=remote
+        ).get("devices")
+        if not devices:
+            # Project "rockcraft" exists but the default profile is ill-formed,
+            # delete the project and start over.
+            raise errors.LXDError(
```

Comment:
Unfortunately deleting the broken projects ourselves involves deleting everything the project is using first, and this is too risky to implement without better investigation and careful testing.